services/jwt-auth/src/main/java/com/gentics/mesh/auth/util/KeycloakUtils.java

```diff
@@ -61,9 +65,24 @@ public static String loadPublicKey(String realmName, String authServerUrl) {
 	 * @throws IOException
 	 */
 	public static Set<JsonObject> loadJWKs(String protocol, String host, int port, String realmName) throws IOException {
+		return loadJWKs(protocol, host, port, realmName, Optional.empty());
+	}
+
+	/**
+	 * Load the JWKs from the realm.
+	 * 
+	 * @param protocol
+	 * @param host
+	 * @param port
+	 * @param realmName
+	 * @param maybeCustomRealmUrlFormat optional realm URL format, suitable for Java formatter with parameters [protocol, host, port, realm name], e.g. '%s://%s:%d/auth/realms/%s'
+	 * @return
+	 * @throws IOException
+	 */
+	public static Set<JsonObject> loadJWKs(String protocol, String host, int port, String realmName, Optional<String> maybeCustomRealmUrlFormat) throws IOException {
```

Comment:
I am not entirely happy with the format of custom Keycloak URLs. While it is ok for internal usage, we expose this to a configuration parameter of the keycloak plugin and this may be error-prone and inconvenient, especially for users who are not familiar with java.

Reply:
As an alternative, this may be implemented as following the Keycloak `realms URL prefix` convention. 
Aside of that I'm totally unhappy of having Keycloak references in an unrelated code.